internal/output/vertical.go

```diff
@@ -11,30 +11,91 @@ import (
+			printVerticalVulnerabilities(source, outputResult.IsContainerScanning, outputWriter)
+			if j < len(ecosystem.Sources)-1 {
+				fmt.Fprintln(outputWriter)
+			}
 		}
 
-		if i < len(vulnResult.Results)-1 {
+		if i < len(outputResult.Ecosystems)-1 {
 			fmt.Fprintln(outputWriter)
 		}
 	}
+
+	fmt.Fprintln(outputWriter)
+}
+
+func printSummary(result Result, out io.Writer) {
+	summary := fmt.Sprintf(
+		"Total %[1]d packages affected by %[2]d vulnerabilities (%[3]s, %[4]s, %[5]s, %[6]s, %[7]s) from %[8]s.\n"+
+			"%[9]d vulnerabilities have fixes available.\n",
+		result.PackageTypeCount.Regular,
+		result.VulnTypeSummary.All,
+		text.FgRed.Sprintf("%d Critical", result.VulnCount.SeverityCount.Critical),
+		text.FgHiYellow.Sprintf("%d High", result.VulnCount.SeverityCount.High),
+		text.FgYellow.Sprintf("%d Medium", result.VulnCount.SeverityCount.Medium),
+		text.FgHiCyan.Sprintf("%d Low", result.VulnCount.SeverityCount.Low),
+		text.FgCyan.Sprintf("%d Unknown", result.VulnCount.SeverityCount.Unknown),
+		text.FgGreen.Sprintf("%d ecosystems", len(result.Ecosystems)),
+		result.VulnCount.FixableCount.Fixed,
+	)
+	fmt.Fprintln(out, summary)
+}
+
+func printBaseImages(imageResult ImageInfo, out io.Writer) {
+	fmt.Fprintf(out, "Container image information:\n")
+	fmt.Fprintf(out, "  OS version: %s\n", text.FgGreen.Sprintf("%s", imageResult.OS))
+	for _, baseImage := range imageResult.AllBaseImages {
+		baseImageString := text.FgYellow.Sprintf("Base Image %d (%s)", baseImage.Index, getBaseImageName(baseImage))
+		if baseImage.Index == 0 {
+			baseImageString = text.FgYellow.Sprintf("Your Image")
+		}
+		fmt.Fprintf(out, "  %s:\n", baseImageString)
+		for _, layer := range baseImage.AllLayers {
+			layerCommand := formatLayerCommand(layer.LayerMetadata.Command)[0]
+			layerCommand = truncate(layerCommand, 100)
+			fmt.Fprintf(out, "    %s %s", text.FgCyan.Sprintf("Layer %d", layer.Index), text.Italic.Sprintf("%s", layerCommand))
+			if layer.Count.AnalysisCount.Regular > 0 {
+				fmt.Fprintf(out, " %s\n", text.FgRed.Sprintf("(%d vulns)", layer.Count.AnalysisCount.Regular))
+			} else {
+				fmt.Fprintln(out)
+			}
+		}
+	}
+	fmt.Fprintln(out)
 }
 
-func printVerticalHeader(result models.PackageSource, out io.Writer) {
+func printVerticalHeader(result SourceResult, out io.Writer) {
 	fmt.Fprintf(
 		out,
 		"%s: found %s %s with issues\n",
-		text.FgMagenta.Sprintf("%s", result.Source.Path),
-		text.FgYellow.Sprintf("%d", len(result.Packages)),
-		Form(len(result.Packages), "package", "packages"),
+		text.FgMagenta.Sprintf("%s", result.Name),
+		text.FgYellow.Sprintf("%d", result.PackageTypeCount.Regular),
+		Form(result.PackageTypeCount.Regular, "package", "packages"),
 	)
 }
 
+func printVerticalPackageContainerInfo(pkg PackageResult, out io.Writer) {
+	baseImageName := getBaseImageName(pkg.LayerDetail.BaseImageInfo)
+	fmt.Fprintf(out, "    Layer introduced in: %s", text.FgCyan.Sprintf("# %d Layer", pkg.LayerDetail.LayerIndex))
```

Comment:
this doesn't seem right to me - we're saying a layer is introduced in a layer?

Should this instead be just "Introduced in: # layer"?

(also I feel like the colon might be redundant if you've got the coloring + `#`, especially since we don't use the `#` when listing the layers above, so I think at least one of those characters should go)

Reply:
Updated it